src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -229,7 +229,7 @@ private void processBatch(Request request, Context context, Promise<Response> oa
           return;
         }
 
-        String nextInstanceId = instances.size() < batchSize ? null : instances.get(batchSize).getString(INSTANCE_ID_FIELD_NAME);
+        String nextInstanceId = instances.size() <= batchSize ? null : instances.get(batchSize).getString(INSTANCE_ID_FIELD_NAME);
```

Comment:
Here was a bug
When the total count of records in a system is equal to batch size the index out of bounds exception is thrown. 
If we have for example 10 records in total and batch size 10 as well, then the first condition will fail and we will try to get id of 11's instance which is absent in the list.